test/scheduler_test.rb

```diff
@@ -457,41 +457,41 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
-  test 'gracefully shuts down rufus-scheduler threads' do
+  test 'finishes enqueueing delayed jobs before shutting down' do
     if RUBY_ENGINE == 'jruby' || RUBY_PLATFORM =~ /mingw|windows/i
       omit("forking is not supported on #{RUBY_ENGINE}/#{RUBY_PLATFORM} but " \
            'this behaviour is best tested using forks')
     end
 
-    class BeforeEnqueueJob
-      @queue = :quick
-
-      class << self
-        def before_enqueue_example(*)
-          return false if enqueue_started?
-          enqueue_started!
-
-          sleep 5
-          true
-        end
+    t = Time.now + 1 # in the future
```

Comment:
this seems kinda racy without `Timecope.freeze`, doesn't it? if it takes more than a second to get to the codepath that looks at this timestamp then this might not be "in the future" anymore. At least use something that's more likely to actually be in the future (like an hour instead of a second)?